app/assets/javascripts/initializers/initializePodcastPlayback.js

```diff
@@ -188,13 +189,11 @@ function initializePodcastPlayback() {
   }
 
   function loadAudio(audio) {
-    if (isNativeIOS()) {
-      sendPodcastMessage({
+    if (isNativePlayer()) {
```

Comment:
A potential idea for the future: introduce a strategy pattern ([example](https://dev.to/carlillo/design-patterns---strategy-pattern-in-javascript-2hg3)) or something similar to only perform this check once, then call the correct strategy (native/non-native) throughout the rest of the code, removing a lot of conditionals in the process.

Reply:
That's a very interesting approach! There definitely are many ways we can optimize and refactor to streamline the logic here. I bookmarked that pattern design series to circle back to it eventually, not just for this specific area 😃

Reply:
@fdoxyz Since our choices seem to be mutually exclusive (web, iOS, or browser) in certain cases, I think we could use this to greatly simplify quite a bit of code. I'd be happy to discuss this in more detail if you ever want to tackle this, maybe during the next downtime or something.